packages/frontend/src/hooks/useQueryResults.tsx

```diff
@@ -202,21 +202,48 @@ export const useChartAndResults = (
     invalidateCache?: boolean,
     granularity?: DateGranularity,
 ) => {
+    const isDateZoomFeatureEnabled = useFeatureFlagEnabled('date-zoom');
+    const queryClient = useQueryClient();
+
     const sortKey =
         dashboardSorts
             ?.map((ds) => `${ds.fieldId}.${ds.descending}`)
             ?.join(',') || '';
-    const queryKey = [
-        'savedChartResults',
-        chartUuid,
-        dashboardFilters,
-        invalidateCache,
-        sortKey,
-        granularity,
-    ];
+    const queryKey = useMemo(
+        () => [
+            'savedChartResults',
+            chartUuid,
+            dashboardFilters,
+            invalidateCache,
+            sortKey,
+            granularity,
+        ],
+        [chartUuid, dashboardFilters, invalidateCache, sortKey, granularity],
+    );
+    const apiChartAndResults = queryClient.getQueryData<
+        ApiChartAndResults & { fetched: boolean }
+    >(queryKey);
+
     const timezoneFixFilters =
         dashboardFilters && convertDateDashboardFilters(dashboardFilters);
 
+    useEffect(() => {
+        if (!isDateZoomFeatureEnabled) return;
+
+        const hasDateDimensions =
+            apiChartAndResults?.metricQuery?.metadata?.hasADateDimension;
+
+        if (hasDateDimensions) {
+            queryClient.invalidateQueries(queryKey, { refetchInactive: true });
```

Comment:
perhaps we could also use the queryKey for this, 

for example: 

```
queryKey = [hasDateDimensions? granularity: undefined] 
```

would it work ?